### ISSUE TITLE
Enhanced Documentation on how Null and NaN values are handled on statistics

### ISSUE THREAD

**Issue by Sage0614** [pola-rs/polars#10125]
### Problem description

currently the arguments that how Null and NaN are handled are quite inconsistent between the statistical functions, some of the functions doesn't provide an option or documentation about how NaN/Null will be handled, and does the handling in some it's own way,

for example, 

pl.var(pl.Series([1,2,3,4,5,float("nan")]))

will give nan where nan is populated;

pl.var(pl.Series([1,2,3,4,5,None]))

will give 2.5 where Null is dropped;

There's no option in govern this behavior in Series aggregation functions,

pl.corr() have an option to govern NaN while pl.cov() does not and so on.

Things like that are quite confusing from user perspective as they are all statistical summary functions from the same library, so people expect the handling, and the options to handle NaN, Inf and Null are somewhat consistent across the whole library. 

So I think it worth to form a consistent interface about options in handling NaN, Inf and Null, like in R, most statistical aggregation functions respect the na.rm arguments:

https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/sum

**Comment by avimallu** [comment]
In Polars, `NaN` and `Null` are very distinct - this is mentioned in the [user guide](https://pola-rs.github.io/polars-book/user-guide/expressions/null/). If you're expecting this behavior from Pandas, note that there is complicated history there with how `NaN` and `Null` values sometimes represent the same things, and sometimes different things. This is because it historically derived its behavior through its dependence on `numpy`, which does not have `Null` values.

As for the lack of clarity on how `Null` values are handled, I agree with you. There is related discussion [in this issue](https://github.com/pola-rs/polars/issues/10016) - please provide your thoughts there :)

**Comment by magarick** [comment]
That's right, you have different kinds of missing-like values. If we can get agreement on how this should actually proceed I'm happy to help work on it.

Personally, I think you need distinct settings for NA and NaN. R will actually ignore both if `na.rm` is `TRUE`. Not only, that but behavior when offending values are found. Do you propagate or throw an error? This is probably a function of whether you're working interactively or in a batch process as well as what you expect of the data. You could also have `_strict` functions for convenience.

**Comment by ion-elgreco** [comment]
Closely related to this issue: https://github.com/pola-rs/polars/issues/10016 (only for null though)
